test/testutils/syncenginetestutils.cpp

```diff
@@ -152,21 +141,25 @@ void FileInfo::remove(const QString &relativePath)
         [&pathComponents](const FileInfo &fi) { return fi.name == pathComponents.fileName(); }));
 }
 
-void FileInfo::insert(const QString &relativePath, qint64 size, char contentChar)
+void FileInfo::insert(const QString &relativePath, quint64 size, char contentChar)
 {
     create(relativePath, size, contentChar);
 }
 
-void FileInfo::setContents(const QString &relativePath, char contentChar)
+void FileInfo::setContents(const QString &relativePath, quint64 newSize, char contentChar)
 {
     FileInfo *file = findInvalidatingEtags(relativePath);
     Q_ASSERT(file);
     file->contentChar = contentChar;
+    file->contentSize = newSize;
+    if (!file->isDehydratedPlaceholder) {
+        file->fileSize = newSize;
+    }
 }
 
 void FileInfo::appendByte(const QString &relativePath, char contentChar)
 {
-    Q_UNUSED(contentChar);
+    Q_UNUSED(contentChar)
```

Comment:
Why is it part of the function signature in the first place?

Reply:
because the function signature of the method in the parent class has it (`FileModifier::appendByte`). The storage on the server side only stores 1 byte, with the assumption that files consist of `contentSize` bytes, all the same, so storing 1 `contentChar` is enough.

I'm not sure what to think of this "design", it's there. See also the FIXMEs in the class definition.